### PR TITLE
Special Number

### LINKED ISSUE #6134: Special Number
## 💥 Proposal

Special number in JAVA

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by Khushi-agarwal**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

It is a JAVA program which is to test whether a number is special or not.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #6134

### CODE REVIEW

**Comment by Khushi-agarwal** [comment]
@HarshCasper kindly review this.

File: Java/math/Special.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+The program is to check whether a given number is a special number or not.
+A special number is a number whose all digits are 1
+*/
+import java.util.*;
+class special
+{
+    public static void main()
```
Comment (HarshCasper) [inline Java/math/Special.java]: Put everything in a modular function. Use `main()` only for the driver code.

File: Java/math/Special.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/*
+The program is to check whether a given number is a special number or not.
+A special number is a number whose all digits are 1
```
Comment (HarshCasper) [inline Java/math/Special.java:3]: Give an example here

File: Java/math/Special.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        while(n!=0)
+        {
+            int b=n%10;
+			/*Calculating sum of digits*/
+            s=s+b;
+			/*Calculating count of digits*/
+            c++;
+            n=n/10;
+        }
+		/*If the sum of digits and count of digits is equal then special*/
```
Comment (HarshCasper) [inline Java/math/Special.java]: Align the comment with the code

File: Java/math/Special.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		/*Taking the number which is to be checked*/
+        System.out.println("Enter the number");
+        int n=sc.nextInt();
+        int s=0,c=0;
+        while(n!=0)
+        {
+            int b=n%10;
+			/*Calculating sum of digits*/
+            s=s+b;
+			/*Calculating count of digits*/
```
Comment (HarshCasper) [inline Java/math/Special.java]: Align the comment with the code

File: Java/math/Special.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        Scanner sc=new Scanner(System.in);
+		/*Taking the number which is to be checked*/
+        System.out.println("Enter the number");
+        int n=sc.nextInt();
+        int s=0,c=0;
+        while(n!=0)
+        {
+            int b=n%10;
+			/*Calculating sum of digits*/
```
Comment (HarshCasper) [inline Java/math/Special.java]: Align the comment with the code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Khushi-agarwal** [comment]
@HarshCasper I have made the changes.Kindly review.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**